src/python_testing/TC_EEVSE_2_3.py

```diff
@@ -0,0 +1,402 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+
+import logging
+from datetime import datetime, timedelta, timezone
+
+import chip.clusters as Clusters
+from chip.clusters.Types import NullValue
+from chip.interaction_model import Status
+from matter_testing_support import EventChangeCallback, MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+from TC_EEVSE_Utils import EEVSEBaseTestHelper
+
+logger = logging.getLogger(__name__)
+
+
+class TC_EEVSE_2_3(MatterBaseTest, EEVSEBaseTestHelper):
+
+    def desc_TC_EEVSE_2_3(self) -> str:
+        """Returns a description of this test"""
+        return "5.1.4. [TC-EEVSE-2.3] Optional ChargingPreferences feature functionality with DUT as Server\n" \
+            "This test case verifies the primary functionality of the Energy EVSE cluster server with the optional ChargingPreferences feature supported."
+
+    def pics_TC_EEVSE_2_3(self):
+        """ This function returns a list of PICS for this test case that must be True for the test to be run"""
+        return ["EEVSE.S", "EEVSE.S.F00"]
+
+    def steps_TC_EEVSE_2_3(self) -> list[TestStep]:
+        steps = [
+            TestStep("1", "Commissioning, already done", is_commissioning=True),
+            TestStep("2", "TH reads TestEventTriggersEnabled attribute from General Diagnostics Cluster. Verify value is 1 (True)"),
+            TestStep("3", "TH sends TestEventTrigger command to General Diagnostics Cluster on Endpoint 0 with EnableKey field set to PIXIT.EEVSE.TEST_EVENT_TRIGGER_KEY and EventTrigger field set to PIXIT.EEVSE.TEST_EVENT_TRIGGER for Basic Functionality Test Event. Verify Command response is Success"),
+            TestStep("4", "TH sends TestEventTrigger command to General Diagnostics Cluster on Endpoint 0 with EnableKey field set to PIXIT.EEVSE.TEST_EVENT_TRIGGER_KEY and EventTrigger field set to PIXIT.EEVSE.TEST_EVENT_TRIGGER for EVSE TimeOfUse Mode Test Event. Verify Command response is Success"),
```

Comment:
```suggestion
            TestStep("4", "TH sends TestEventTrigger command to General Diagnostics Cluster on Endpoint 0 with EnableKey field set to PIXIT.EEVSE.TEST_EVENT_TRIGGER_KEY and EventTrigger field set to PIXIT.EEVSE.TEST_EVENT_TRIGGER for EVSE TimeOfUse Mode Test Event.", "Verify Command response is Success"),
```

Reply:
The same change can be made to all of these TestSteps now that there is a 3rd argument which is the test verification step.

It helps make the file easier to read rather than long lines of text. Once the Python formatter has nicely done line wrapping.

Reply:
Applied the change now - this can be resolved.